From 2d31e373d5345966604f773220f94a703fc7af30 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 16 Jul 2021 06:05:53 +0300
Subject: [PATCH] [SW-44497] greco: ts alarm threshold update

Alarm a and b thresholds are updated in preboot, aligning with zephyr
pvt driver.

Thresholds(in celsius):
existing:
alarm_a = 102 , alarm_b = 120
updated:
alarm_a = 90 , alarm_b = 115

The thresholds are updated for all 4 instances of ts(DCORE0,DCON0,
DCORE1,DCON2).

Change-Id: I9770a2d5cb9edacd5ce23ad42041adef7b54c65a
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
---

diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index 68e2e36..ede9a67 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -324,13 +324,13 @@
 }
 
 static struct hl_ts_cfg greco_hl_ts_data[] = {
-	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 90, .alarm_b = 115,
 		.use_calib = false, .calib_offset = 10, .hl_tmp_data = { 0 }},
-	{.base = mmDCON0_TSTD_BASE, .alarm_a = 102, .alarm_b =  120,
+	{.base = mmDCON0_TSTD_BASE, .alarm_a = 90, .alarm_b =  115,
 		.use_calib = false, .calib_offset = 9, .hl_tmp_data = { 0 }},
-	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 90, .alarm_b = 115,
 		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }},
-	{.base = mmDCON2_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+	{.base = mmDCON2_TSTD_BASE, .alarm_a = 90, .alarm_b = 115,
 		.use_calib = false, .calib_offset = 7, .hl_tmp_data = { 0 }}
 };
 
diff --git a/arch/greco/arm/common.c b/arch/greco/arm/common.c
index 8ade6c1..a35b106 100644
--- a/arch/greco/arm/common.c
+++ b/arch/greco/arm/common.c
@@ -753,13 +753,13 @@
 }
 
 static struct hl_ts_cfg greco_hl_ts_data[] = {
-	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 90, .alarm_b = 115,
 		.use_calib = false, .calib_offset = 10, .hl_tmp_data = { 0 }},
-	{.base = mmDCON0_TSTD_BASE, .alarm_a = 102, .alarm_b =  120,
+	{.base = mmDCON0_TSTD_BASE, .alarm_a = 90, .alarm_b =  115,
 		.use_calib = false, .calib_offset = 9, .hl_tmp_data = { 0 }},
-	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 90, .alarm_b = 115,
 		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }},
-	{.base = mmDCON2_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+	{.base = mmDCON2_TSTD_BASE, .alarm_a = 90, .alarm_b = 115,
 		.use_calib = false, .calib_offset = 7, .hl_tmp_data = { 0 }}
 };
 
